node/src/components/block_accumulator/block_acceptor.rs

```diff
@@ -196,64 +199,123 @@ impl BlockAcceptor {
         &mut self,
         era_validator_weights: &EraValidatorWeights,
     ) -> (ShouldStore, Vec<(NodeId, AcceptorError)>) {
-        if self.has_sufficient_finality {
+        let block_hash = self.block_hash;
+        let no_block = self.hot_block.is_none();
+        let no_sigs = self.signatures.is_empty();
+        if self.has_sufficient_finality() {
+            if let Some(hot_block) = self.hot_block.as_mut() {
+                if hot_block.state.is_executed()
+                    && hot_block.state.register_as_marked_complete().was_updated()
+                {
+                    debug!(
+                        %block_hash, no_block, no_sigs,
+                        "already have sufficient finality signatures, but marking block complete"
+                    );
+                    return (ShouldStore::MarkComplete(hot_block.clone()), Vec::new());
+                }
+            }
+
             debug!(
-                block_hash = %self.block_hash,
-                no_block = self.block.is_none(),
-                no_sigs = self.signatures.is_empty(),
+                %block_hash, no_block, no_sigs,
                 "not storing anything - already have sufficient finality signatures"
             );
             return (ShouldStore::Nothing, Vec::new());
         }
 
-        if self.block.is_none() || self.signatures.is_empty() {
-            debug!(
-                block_hash = %self.block_hash,
-                no_block = self.block.is_none(),
-                no_sigs = self.signatures.is_empty(),
-                "not storing block"
-            );
+        if self.hot_block.is_none() || self.signatures.is_empty() {
```

Comment:
Nit:

```suggestion
        if no_block || no_sigs {
```

Reply:
Fixed in [e4d8cc6](https://github.com/casper-network/casper-node/pull/3549/commits/e4d8cc609a03671e5923825bf06174f6c64e9e7f).